react/src/components/lablupTalkativotUI/LLMChatCard.tsx

```diff
@@ -160,12 +165,41 @@ const LLMChatCard: React.FC<LLMChatCardProps> = ({
     }
   }, [inputMessage, setInput]);
 
+  // If the `inputAttachment` prop exists, the `files` state has to follow it.
+  useEffect(() => {
+    if (!_.isUndefined(inputAttachment) && !_.isEqual(files, inputAttachment)) {
+      setFiles(inputAttachment);
+      setIsOpenAttachments(true);
+    }
+  }, [inputAttachment, files]);
+
+  const createDataTransferFiles = (files: AttachmentsProps['items']) => {
+    const fileList = _.map(files, (item) => item.originFileObj as File);
+    const dataTransfer = new DataTransfer();
+    _.forEach(fileList, (file) => {
+      dataTransfer.items.add(file);
+    });
+    return dataTransfer.files;
+  };
+
   useEffect(() => {
     if (!_.isUndefined(submitKey) && input) {
-      append({
-        role: 'user',
-        content: input,
-      });
+      if (!_.isEmpty(files)) {
```

Comment:
In case of calling same function with conditional props in options, we can make it more clear to separate it into two of phases.

// step 1, setup options

const chatRequestOptions: ChatRequestOptions = {
  // setup option with condition
};

// step 2, call a function once

append({
          role: 'user',
          content: input,
}, chatRequestOptions);